src/open_company/resources/company.clj

```diff
@@ -25,37 +27,28 @@
 (defn- categories-for
   "Add the :categories vector in the company with the definitive initial list of categories."
   [company]
-  (assoc company :categories common/categories))
+  (assoc company :categories (vec common/category-names)))
 
 (defn- section-list
   "Return the set of section names that are contained in the provided company."
   [company]
-  (set (clojure.set/intersection common/sections (set (keys company)))))
+  (set (cset/intersection common/section-names (set (keys company)))))
 
-(defun- sections-for
-  "Add a :sections vector in the new company with the sections that this company map actually contains."
-
-  ; init
-  ([company] (sections-for company (keys common/ordered-sections) (section-list company) {}))
-
-  ; no more categories
-  ([company _categories :guard empty? _sections section-matches] (assoc company :sections section-matches)) ; all done
-
-  ; section matches per category
-  ([company categories sections section-matches]
-  (let [category-name (first categories)
-        category-sections (common/ordered-sections category-name)
-        matches-for-category (filter #(some sections [%]) category-sections)]
-    (sections-for company
-      (rest categories)
-      sections
-      (assoc section-matches category-name (vec matches-for-category))))))
+(defn sections-for
+  "Add a :sections key to given company containing category->ordered-sections mapping
+  Only add sections to the ordered-sections list that are used in the company map"
+  [company]
+  (let [seclist (section-list company)]
+    (->> (for [[cat sects] common/category-section-tree]
+          [cat (vec (filter (set seclist) sects))])
+      (into {})
+      (assoc company :sections))))
 
 (defn- remove-sections
   "Remove any sections from the company that are not in the :sections property"
   [company]
   (let [sections (set (map keyword (flatten (vals (:sections company)))))]
-    (apply dissoc company (clojure.set/difference (section-list company) sections))))
+    (apply dissoc company (cset/difference (section-list company) sections))))
 
 (defn sections-with
```

Comment:
It's used in `sections.clj` (only there) but I didn't touch it in this PR so I didn't see until now. Should I just move it there?